### PR TITLE
feat: add LDAP support

### PR DESCRIPTION
**Pull request by baznikin**
Application runs in container as non-root user and can't bind to lower port.
This PR adds ability to change LDAP port and expose it with Service.
Access to LDAP port from outside of K8s cluster is out of scope of this PR

### CODE REVIEW

**Comment by hsluoyz** [comment]
@sp71 @MichaelSp plz review

**Comment by baznikin** [comment]
> @baznikin why so many code lines are changed?

There are some minor imperfection - empty lines in generated YAML files. At first I decided to keep formating same way as you do, but later I noticed you simply miss `-` in some places and put it in others. So, I decide to fix formatting as well. Sorry, I forget to mention it

**Comment by hsluoyz** [comment]
Formatting should be in separated PRs. One PR one thing

**Comment by baznikin** [comment]
> Formatting should be in separated PRs. One PR one thing

You are right, fixed

File: charts/casdoor/values.yaml [hunk 1]
```diff
@@ -46,6 +47,11 @@ database:
 
   sslMode: disable
 
+ldap:
+  enabled: false
+  service:
+    port: 10389
```
Comment (hsluoyz) [inline charts/casdoor/values.yaml:53]: default is 389

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: charts/casdoor/values.yaml (hunk 1 above)
Comment (baznikin) [reply charts/casdoor/values.yaml:53]: Yes, that's why it's not worked in container. Only root can bind to lower ports (<1024).
Let's bind inside container on some port of high range, and set port in Service to configurable value with default 139?

File: charts/casdoor/templates/deployment.yaml [hunk 2]
```diff
@@ -56,7 +56,12 @@ spec:
             - name: http
               containerPort: {{ .Values.service.port }}
               protocol: TCP
-          {{ if .Values.probe.liveness.enabled }}
+            {{- if .Values.ldap.enabled }}
+            - name: ldap
+              containerPort: {{ .Values.ldap.service.port }}
+              protocol: TCP
+            {{- end }}
+          {{- if .Values.probe.liveness.enabled }}
```
[the remark this answers is not in the document]
Comment (baznikin) [reply charts/casdoor/templates/deployment.yaml:64]: missed this place when rollback format changes

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**